utils/logger.go

```diff
@@ -0,0 +1,75 @@
+package utils
```

Comment:
Sorry. the PR is huge so maybe I'm missing something...

Do we really need this interface?